### PR TITLE
Add PBI medicare dataset & vortex/parquet comparison

### PR DESCRIPTION
**Pull request by jdcasale**
- Adds one of the medicare datasets* from the ALP paper to our compress and random_access benchmarks
- Adds mechanism for handling csv data
- Adds mechanism for handling bzipped data
- Refactors to enable translating general parquet to lance 
- General refactors to make data download/translation ops more modular

*https://github.com/cwida/public_bi_benchmark/tree/master/benchmark/Medicare1 -- contains two bzipped files, this downloads just the first

### CODE REVIEW

File: bench-vortex/src/medicare_data.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+use crate::data_downloads::{decompress_bz2, download_data, parquet_to_lance};
+use crate::idempotent;
+use crate::reader::{compress_csv_to_vortex, default_csv_format, write_csv_as_parquet};
+
+pub fn medicare_data_csv() -> PathBuf {
+    let fname = "Medicare1_1.csv.bz2";
+    download_data(
+        fname,
+        "http://www.cwi.nl/~boncz/PublicBIbenchmark/Medicare1/Medicare1_1.csv.bz2",
```
Comment (lwwmanning) [inline bench-vortex/src/medicare_data.rs:24]: FLUP, but we should probably centralize our testing data in a place we own (maybe just a public S3/R2 bucket)

File: bench-vortex/src/medicare_data.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        let mut write = File::create(path).unwrap();
+        let mut write_ctx = WriteCtx::new(&mut write);
+        write_ctx.dtype(&dtype)?;
+        write_ctx.write(&chunked)
+    })
+    .unwrap()
+}
+
+pub fn medicare_data_vortex() -> PathBuf {
+    idempotent("taxi.vortex", |path| {
```
Comment (lwwmanning) [inline bench-vortex/src/medicare_data.rs]: wrong file name

File: bench-vortex/src/medicare_data.rs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+pub fn medicare_data_lance() -> PathBuf {
+    let taxi_data = File::open(medicare_data_parquet()).unwrap();
+    idempotent("taxi.lance", |path| {
+        Ok::<PathBuf, VortexError>(parquet_to_lance(path, taxi_data))
+    })
+    .unwrap()
+}
+
+pub fn medicare_data_vortex_uncompressed() -> PathBuf {
+    idempotent("taxi-uncompressed.vortex", |path| {
```
Comment (lwwmanning) [inline bench-vortex/src/medicare_data.rs]: wrong file name

File: vortex-fastlanes/Cargo.toml [hunk 4]
```diff
@@ -16,6 +16,7 @@ workspace = true
 
 [dependencies]
 arrayref = { workspace = true }
+parquet = { workspace = true }
```
Comment (lwwmanning) [inline vortex-fastlanes/Cargo.toml]: why does fastlanes depend on parquet? this looks vestigial

File: bench-vortex/src/medicare_data.rs (hunk 1 above)
Comment (jdcasale) [reply bench-vortex/src/medicare_data.rs:24]: Yeah I agree that we should just grab a copy of all of the public BI benchmark data and host it ourselves. That way if their website is down and/or goes offline and/or changes we have a canonical copy.

File: bench-vortex/src/medicare_data.rs (hunk 2 above)
Comment (jdcasale) [reply bench-vortex/src/medicare_data.rs]: whoops

File: vortex-fastlanes/Cargo.toml (hunk 4 above)
Comment (jdcasale) [reply vortex-fastlanes/Cargo.toml]: You're right -- this is vestigial

File: bench-vortex/src/medicare_data.rs [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    download_data(
+        fname,
+        "http://www.cwi.nl/~boncz/PublicBIbenchmark/Medicare1/Medicare1_1.csv.bz2",
+    );
+    decompress_bz2(fname, "Medicare1_1.csv")
+}
+
+pub fn medicare_data_lance() -> PathBuf {
+    let taxi_data = File::open(medicare_data_parquet()).unwrap();
+    idempotent("taxi.lance", |path| {
```
Comment (lwwmanning) [inline bench-vortex/src/medicare_data.rs]: one more :)

**Review by lwwmanning — APPROVED** [review]

File: bench-vortex/src/medicare_data.rs (hunk 5 above)
Comment (jdcasale) [reply bench-vortex/src/medicare_data.rs]: sorry, my bad

**Pull request merged**